src/Raven.Client/Documents/Operations/Replication/BucketMigrationReplication.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Replication;
+using Raven.Client.ServerWide;
+using Sparrow;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.Replication
+{
+    public class BucketMigrationReplication : ReplicationNode
+    {
+        public int Bucket;
+        public int Shard;
+        public readonly string Node;
+        public readonly long MigrationIndex;
+
+        public BucketMigrationReplication(string node, long migrationIndex)
+        {
+            Node = node ?? throw new ArgumentNullException(nameof(node));
+            MigrationIndex = migrationIndex;
+        }
+
+        public bool ForBucketMigration(BucketMigration migration)
+        {
+            if (migration.MigrationIndex != MigrationIndex)
```

Comment:
Can we have different migration index on same bucket / shard ?

Reply:
Yes, but not concurrently.
consider  
 - move bucket 123 from 0 to 1 at index 10
 - move bucket 123 from 1 to 0 at index 20
 - move bucket 123 from 0 to 3 at index 30

first and last has same shard/bucket but different index

Reply:
So why do we need the `MigrationIndex` check?